src/node_i18n.cc

```diff
@@ -64,7 +79,104 @@ bool InitializeICUDirectory(const char* icu_data_path) {
   }
 }
 
+static void ToUnicode(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  CHECK_GE(args.Length(), 1);
+  CHECK(args[0]->IsString());
+  Utf8Value val(env->isolate(), args[0]);
+  UErrorCode status = U_ZERO_ERROR;
+  uint32_t options = UIDNA_DEFAULT;
+  UIDNA* uidna = uidna_openUTS46(options, &status);
+  UIDNAInfo info = UIDNA_INFO_INITIALIZER;
+
+  char* dest;
+  char buf[1024];
+  dest = buf;
+  int32_t len = uidna_nameToUnicodeUTF8(uidna,
+                                        *val, val.length(),
+                                        dest, 1024,
+                                        &info,
+                                        &status);
+
+  if (status == U_BUFFER_OVERFLOW_ERROR) {
+    status = U_ZERO_ERROR;
+    dest = static_cast<char*>(malloc(len));
+    len = uidna_nameToUnicodeUTF8(uidna,
+                                  *val, val.length(),
+                                  dest, len,
+                                  &info,
+                                  &status);
+  }
+
+  if (U_FAILURE(status)) {
+    env->ThrowError("Cannot convert name to unicode");
+    goto clean;
+  }
+
+  args.GetReturnValue().Set(
+    String::NewFromUtf8(env->isolate(), dest, String::kNormalString, len));
```

Comment:
hmm.. sorry, not clear, which version is the non-deprecated one? quick search through src shows this form being used in several places so it's not exactly clear (sorry, just missing something on my end ;-) ...)